src/main/java/br/unb/cloudissues/Main.java

```diff
@@ -1,194 +1,185 @@
+
+		requestAndWriteFalsePositiveAndWontFixViolations(PROJECTS_LIST,
+				WONT_FIX_FALSE_POSITIVE_DIRECTORY);
+
+		projectsAndFilesMetrics();
+	}
+
+	private static void retrieveAndWriteRules() throws IOException, InterruptedException {
+		JavaRulesRetriever jrr = new JavaRulesRetriever(SONAR_API_URL);
+		List<Rule> rules = jrr.retrieve();
+		Utils.writeObjToFileAsJSON(rules, RULES_LIST);
+	}
+
+	private static void retrieveAndWriteProjects() throws IOException, InterruptedException {
+		JavaProjectsRetriever jpr = new JavaProjectsRetriever.Builder(SONAR_API_URL)
+				.isSonarCloud(false).build();
+		List<Project> sonarJavaProjects = jpr.retrieve();
+		Utils.writeObjToFileAsJSON(sonarJavaProjects, PROJECTS_LIST);
+	}
+
+	static void retrieveAndWriteProjectsIfApache() throws IOException {
+		JavaProjectsCSVRetriever jpr = new JavaProjectsCSVRetriever(SONAR_API_URL + "/projects");
+		Utils.writeObjToFileAsJSON(jpr.retrieve(), PROJECTS_LIST);
+	}
+
+	private static void requestAndWriteFixedViolationsOneFilePerProject(String projectsJsonPath,
+			String fixedDirectory) throws IOException, InterruptedException {
+		System.out.println("\nRetrieving fixed violations");
+
+		List<Project> projects = Utils.retrieveCollectionFromJSONFile(projectsJsonPath,
+				Project.class);
+
+		for (Project project : projects) {
+			requestAndWriteFixedViolationsForProjects(Utils.generateJsonPathToSaveForEachProject(
+					fixedDirectory, project.getProjectName()), Collections.singletonList(project));
+			waitBetweenPartitions(10_000);
+		}
+	}
+
+	private static void requestAndWriteFixedViolationsForProjects(String jsonPathToSave,
+			List<Project> projects) throws IOException {
+		ViolationsRetriever violationsRetriever = new ViolationsRetriever.Builder()
+				.withIsSonarCloud(false) //
+				.withBaseUrl(ISSUES_SEARCH_URL) //
+				.withMaxRequestsToWait(15) //
+				.withTimeToWaitInMiliSecondsBetweenRequests(10_000)
+				.withResolutions(Resolutions.FIXED) //
+				.build();
+
+		List<Violations> viols = violationsRetriever.retrieve(projects);
+
+		Utils.writeObjToFileAsJSON(viols, jsonPathToSave);
+	}
+
+	private static void waitBetweenPartitions(int timeInMiliSeconds) throws InterruptedException {
+		System.out.println(
+				"\nwaiting " + (timeInMiliSeconds / 1000) + " seconds between partitions...");
+		Thread.sleep(timeInMiliSeconds);
+	}
+
+	private static void requestAndWriteOpenViolationsOneFilePerProject(String projectsJsonPath,
+			String openDirectory) throws IOException, InterruptedException {
+		System.out.println("\nRetrieving open violations");
+
+		List<Project> projects = Utils.retrieveCollectionFromJSONFile(projectsJsonPath,
+				Project.class);
+
+		for (Project project : projects) {
+			requestAndWriteOpenViolationsForProjects(Utils.generateJsonPathToSaveForEachProject(
+					openDirectory, project.getProjectName()), Collections.singletonList(project));
+			waitBetweenPartitions(10_000);
```

Comment:
#### Changes Here
 Changing the operator asList to singleList, the motivation is because: 
     the asList is mutable and the operation from methods `requestAndWriteOpenViolationsForProjects` and `requestAndWriteFalsePositiveViolationsForProjects` 
     not handle(inserting, removing, and change, just consuming values)